FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/MsgDetailsViewModel.kt

```diff
@@ -476,8 +477,16 @@ class MsgDetailsViewModel(
             Result.success(null)
           }
           if (result.status == Result.Status.SUCCESS) {
-            return@withContext MsgsCacheManager.getMsgSnapshot(messageEntity.id.toString())
-              ?: throw java.lang.NullPointerException("Message not found in the local cache")
+            var snapshot = MsgsCacheManager.getMsgSnapshot(messageEntity.id.toString())
+            if (snapshot == null) {
+              //due to realization of [DiskLruCache] we need to do a little delay
+              // while [DiskLruCache] is updating internal data
+              delay(500)
+              snapshot = MsgsCacheManager.getMsgSnapshot(messageEntity.id.toString())
+            }
```

Comment:
Maybe could be rewritten as do-while or similar? Or `for 1..50 { if (!null) return ...; delay ... } throw ...`

Then you set the interval at 50ms and try up to 50 times, which will be both faster (usually settles in 50ms, probably) and more reliable (waits up to 2.5 seconds instead of 0.5 to throw error)